src/AbsMSeries.jl

```diff
@@ -12,18 +12,18 @@
 ###############################################################################
 
 function O(a::AbsMSeriesElem{T}) where T <: RingElement
-    if iszero(a)
-       return deepcopy(a)
-    end
-    R = parent(a)
-    p = poly(a)
-    v = vars(p)
-    (length(v) != 1 || length(p) != 1 || !isone(leading_coefficient(p))) &&
-                                               error("Not a pure power in O()")
-    ind = var_index(v[1])
-    exps = first(exponent_vectors(p))
-    prec = [i == ind ? exps[i] : R.prec_max[i] for i in 1:length(exps)]
-    return R(parent(p)(), prec)
+  if iszero(a)
+    return deepcopy(a)
+  end
+  R = parent(a)
+  p = poly(a)
+  v = vars(p)
+  (length(v) != 1 || length(p) != 1 || !isone(leading_coefficient(p))) &&
+  error("Not a pure power in O()")
```

Comment:
This does not look.

Reply:
I made an `if` out of this. It took me a moment to understand what these `||` and `&&` do.